src/Fixer/ControlStructure/ControlStructureContinuationFixer.php

```diff
@@ -0,0 +1,222 @@
+
+            if (!$prevToken->equals('}')) {
+                continue;
+            }
+
+            $tokens->ensureWhitespaceAtIndex(
+                $index - 1,
+                1,
+                self::NEXT_LINE === $this->configuration['keyword_position'] ?
+                    $this->whitespacesConfig->getLineEnding().$this->detectIndent($tokens, $index)
+                    : ' '
+            );
+        }
+    }
+
+    private function detectIndent(Tokens $tokens, int $index): string
+    {
+        while (true) {
+            $whitespaceIndex = $tokens->getPrevTokenOfKind($index, [[T_WHITESPACE]]);
+
+            if (null === $whitespaceIndex) {
+                return '';
+            }
+
+            $whitespaceToken = $tokens[$whitespaceIndex];
+
+            if (false !== strpos($whitespaceToken->getContent(), "\n")) {
+                break;
+            }
+
+            $prevToken = $tokens[$whitespaceIndex - 1];
+
+            if ($prevToken->isGivenKind([T_OPEN_TAG, T_COMMENT]) && "\n" === substr($prevToken->getContent(), -1)) {
+                break;
+            }
+
+            $index = $whitespaceIndex;
+        }
+
+        $explodedContent = explode("\n", $whitespaceToken->getContent());
+
+        return end($explodedContent);
+    }
+
+    private function fixDoWhile(Tokens $tokens): void
+    {
+        for ($index = \count($tokens) - 1; 0 <= $index; --$index) {
+            $token = $tokens[$index];
+
+            if (!$token->isGivenKind(T_DO)) {
+                continue;
+            }
+
+            $parenthesisEndIndex = $this->findParenthesisEnd($tokens, $index);
+            $startBraceIndex = $tokens->getNextNonWhitespace($parenthesisEndIndex);
+            $endBraceIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $startBraceIndex);
+            $nextNonWhitespaceIndex = $tokens->getNextNonWhitespace($endBraceIndex);
+            $nextNonWhitespaceToken = $tokens[$nextNonWhitespaceIndex];
+
+            if (!$nextNonWhitespaceToken->isGivenKind(T_WHILE)) {
+                continue;
+            }
+
+            $tokens->ensureWhitespaceAtIndex(
+                $nextNonWhitespaceIndex - 1,
+                1,
+                self::NEXT_LINE === $this->configuration['keyword_position'] ?
+                    $this->whitespacesConfig->getLineEnding().$this->detectIndent($tokens, $index)
+                    : ' '
+            );
+        }
+    }
+
+    private function findParenthesisEnd(Tokens $tokens, int $structureTokenIndex): int
+    {
+        $nextIndex = $tokens->getNextMeaningfulToken($structureTokenIndex);
+        $nextToken = $tokens[$nextIndex];
+
+        // return if next token is not opening parenthesis
+        if (!$nextToken->equals('(')) {
+            return $structureTokenIndex;
+        }
+
+        return $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $nextIndex);
+    }
+
+    private function getControlContinuationTokens()
```

Comment:
Can we replace this function with private constant?

Reply:
Done.